Source/WebKit/webpushd/WebPushDaemon.mm

```diff
@@ -533,7 +532,36 @@ static void tryCloseRequestConnection(xpc_object_t request)
 void WebPushDaemon::handleIncomingPushImpl(const PushSubscriptionSetIdentifier& identifier, WebKit::WebPushMessage&& message)
 {
     ensureIncomingPushTransaction();
+
+#if ENABLE(DECLARATIVE_WEB_PUSH)
+    String payloadString;
+    if (message.pushData)
+        payloadString = String::fromUTF8(message.pushData->span());
+
+    if (!payloadString.isNull()) {
+        auto exceptionOrPayload = WebCore::NotificationPayload::parseJSON(payloadString);
+        if (!exceptionOrPayload.hasException())
+            message.notificationPayload = exceptionOrPayload.returnValue();
+    }
+
+#if HAVE(FULL_FEATURED_USER_NOTIFICATIONS)
```

Comment:
This does not seem right for macOS Safari since we need to defer to SafariNotificationAgent to show the notification using the legacy web notification center.

Reply:
Yup, handling this differently.